src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.js

```diff
@@ -174,40 +192,55 @@ export default function CreateOfferRequest ({ history: { push } }) {
         name='notes'
         placeholder='What is this for?'
         ref={register} />
+
       <RecentCounterparties
         styleName='recent-counterparties'
         agents={agents}
         selectedAgentId={counterpartyId}
         selectAgent={selectAgent} />
-      <Button type='submit' wide variant='secondary' styleName='send-button' dataTestId='submit-button'>
-        {modeVerbs[mode]}
-      </Button>
+      <Button type='submit' wide variant='secondary' styleName='send-button' disabled={counterpartyId.length === AGENT_ID_LENGTH ? counterpartyNotFound : true}>Send</Button>
     </form>
   </PrimaryLayout>
 }
 
-export function RenderNickname ({ agentId, setCounterpartyNick }) {
-  const { loading, error, data: { holofuelCounterparty = {} } = {} } = useQuery(HolofuelCounterpartyQuery, {
+export function RenderNickname ({ agentId, setCounterpartyNick, setErrorMessage, errorMessage, setCounterpartyNotFound, counterpartyNotFound, currentCounterpartyValue }) {
+  const { loading, error: queryError, data: { holofuelCounterparty = {} } = {} } = useQuery(HolofuelCounterpartyQuery, {
     variables: { agentId }
   })
   const { nickname } = holofuelCounterparty
   useEffect(() => {
     setCounterpartyNick(nickname)
   }, [setCounterpartyNick, nickname])
 
+  if (!loading && holofuelCounterparty.notFound) {
+    errorMessage = 'This HoloFuel Peer is currently unable to be located in the network. \n Please verify the hash, ensure your HoloFuel Peer is online, and try again after a few minutes.'
+    counterpartyNotFound = true
+  } else if (!loading && !errorMessage && currentCounterpartyValue && currentCounterpartyValue.length === AGENT_ID_LENGTH) {
+    counterpartyNotFound = false
+  } else if (loading) counterpartyNotFound = true
+
+  useEffect(() => {
+    setErrorMessage(errorMessage)
+  }, [setErrorMessage, errorMessage])
+
+  useEffect(() => {
+    setCounterpartyNotFound(counterpartyNotFound)
+  }, [setCounterpartyNotFound, counterpartyNotFound])
+
   if (loading) {
+    // TODO: Unsubscribe from Loader to avoid any potential mem leak.
```

Comment:
Can you help me understand this comment. afaict, if `(!loading)`, `Loader` will be removed from the dom by React in the normal way, no danger of memory leaks.

Reply:
Let's look at this together. There is actually a React Warning in the console that states such.  I'll show you when we pair. :eyes: